sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -211,50 +216,44 @@ public void AddTag(string name, object value)
                 }
             }
 
-            private List<ActivityLink>? GetActivitySourceLinkCollection()
+            private IEnumerable<Activity> GetDiagnosticSourceLinkCollection()
             {
                 if (_links == null)
                 {
-                    return null;
+                    return Array.Empty<Activity>();
                 }
 
-                var linkCollection = new List<ActivityLink>();
+                var linkCollection = new List<Activity>();
 
-                foreach (var activity in _links)
+                foreach (var link in _links)
                 {
-                    ActivityTagsCollection linkTagsCollection = new();
-                    foreach (var tag in activity.Tags)
+                    var activity = new Activity("LinkedActivity");
+                    if (link.Context != default) // TODO
```

Comment:
Is the // TODO comment intentional? If so, what is left to do? :)

Reply:
unintentional - left from the time there was something to do :)